src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/EnumShouldNotHaveDuplicatedValues.cs

```diff
@@ -0,0 +1,150 @@
+                                                                                            customTags: WellKnownDiagnosticTags.Telemetry);
+
+        private static readonly LocalizableString s_localizableMessageRuleDuplicatedBitwiseValuePart = new LocalizableResourceString(nameof(MicrosoftCodeQualityAnalyzersResources.EnumShouldNotHaveDuplicatedValuesMessageDuplicatedBitwiseValuePart), MicrosoftCodeQualityAnalyzersResources.ResourceManager, typeof(MicrosoftCodeQualityAnalyzersResources));
+        internal static DiagnosticDescriptor RuleDuplicatedBitwiseValuePart = new DiagnosticDescriptor(RuleId,
+                                                                                                       s_localizableTitle,
+                                                                                                       s_localizableMessageRuleDuplicatedBitwiseValuePart,
+                                                                                                       DiagnosticCategory.Design,
+                                                                                                       DiagnosticHelpers.DefaultDiagnosticSeverity,
+                                                                                                       isEnabledByDefault: false,
+                                                                                                       helpLinkUri: null,
+                                                                                                       customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public sealed override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(RuleDuplicatedValue, RuleDuplicatedBitwiseValuePart);
+
+        public sealed override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterSymbolStartAction(ssac =>
+            {
+                var enumSymbol = (INamedTypeSymbol)ssac.Symbol;
+                if (enumSymbol.TypeKind != TypeKind.Enum)
+                {
+                    return;
+                }
+
+                var invalidMembersByValue = new ConcurrentDictionary<object, List<SyntaxNode>>();
+
+                // Collect duplicated values...
+                ssac.RegisterOperationAction(oc =>
+                {
+                    var fieldInitializerOperation = (IFieldInitializerOperation)oc.Operation;
+                    if (fieldInitializerOperation.InitializedFields.Length != 1 ||
+                        !fieldInitializerOperation.Value.ConstantValue.HasValue)
+                    {
+                        return;
+                    }
+
+                    var onlyReferencesOneField = GetFilteredDescendants(fieldInitializerOperation, op => op.Kind != OperationKind.Binary)
```

Comment:
Filters referenced fields that are not part of a binary expression. If any there shouldn't be more than 1 found, do you want me to use a debug specific code with an assertion for this?

Reply:
What if the referenced field belongs to a different type? Do you need to check the containing type of referenced fields?